pkg/csource/options.go

```diff
@@ -25,10 +25,9 @@ type Options struct {
 	Procs       int    `json:"procs"`
 	Slowdown    int    `json:"slowdown"`
 	Sandbox     string `json:"sandbox"`
-
-	Fault     bool `json:"fault,omitempty"` // inject fault into FaultCall/FaultNth
-	FaultCall int  `json:"fault_call,omitempty"`
-	FaultNth  int  `json:"fault_nth,omitempty"`
+	// This option is still present to be able to restrict fault injection only to thoses OSes
+	// which support it.
```

Comment:
Hmmm, yes, it seems that it lost much of its purpose during one of the redevelopments of this PR.

But still. If we move it to legacy right now, two issues would arise:
* Theoretically one could write a program that uses fault injection as a call property and pass it to `syz-prog2c`. If the OS does not support fault injection, the generated C file just would not compile. Is that ok?
* We also want to test, whether enabled fault injection results in compilable C programs (when it's ok for it to be enabled). In this PR it's done by the `testTarget` function inside `pkg/csource/csource_test.go`.
If the knowledge about whether fault injection is supported by a target is not stored along with similar logic inside `pkg/csource/options.go`, we'd have to just hard code it in the test. Is that fine?

Reply:
> Theoretically one could write a program that uses fault injection as a call property and pass it to syz-prog2c. If the OS does not support fault injection, the generated C file just would not compile. Is that ok?

I can't think of any cases where we take program for one OS and deserialize it as a program for another OS. This can fail for lots of other reasons (syscalls are different to begin with).
But let's say we don't want to fail badly on this case. Using opts.Fault looks strange for this purpose. What if I (caller of pkg/csource) simply sets opts.Fault for OS that does not support it. Or what if the program contains fault injection, but opts.Fault is not set? Or the other way around?

The program is now the source of truth, duplicating fault in opts looks like it only adds problems.
So could we use prog.HasFaultInjection and then check somewhere that the program does not contain fault injection if OS does not support it?

Reply:
> What if I (caller of pkg/csource) simply sets opts.Fault for OS that does not support it.

Then it won't pass `Options.Check` and an error will be printed.

> Or what if the program contains fault injection, but opts.Fault is not set?

In this PR fault injection props are just ignored in this case.

> Or the other way around?

It also won't be failing in this PR.

> So could we use prog.HasFaultInjection and then check somewhere that the program does not contain fault injection if OS does not support it?

Theoretically we could make `prog.validate` check fault injection support and rename it to `prog.Validate`, so that it can be used more widely. This should both solve `prog2c` issues and let us not duplicate fault injection support checking in each test. What do you think?

Reply:
> Then it won't pass Options.Check and an error will be printed.

What I am getting at is that it's a properly of the target, not something for csource caller to decide/set. If target is Linux it is supported. Period. If it's FreeBSD, it's not supported. Period. It does not make sense as user settable thing in csource.Options.

> Theoretically we could make prog.validate check fault injection support and rename it to prog.Validate

prog.Deserialize already calls validate, so it should work as is.

Reply:
Moved the field to LegacyOptions and added fault injection stubs to executor, so that it always builds.